### PR TITLE
#13283 Resolve @NeedsTest Issues for flagToDisplay in Reviewer.kt

### PR DESCRIPTION
**Pull request by taulpi**
## Purpose / Description
I chose a few @NeedsTest Annotations and covered the behaviour with Unit Tests

## Fixes
#13283

## Approach
I separated the logic-related code from the view-related code in the FlagToDisplay class and covered it with tests.

## How Has This Been Tested?
I ran './gradlew jacocoTestReport' and made sure that my the code is covered in the report

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/FlagToDisplay.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/****************************************************************************************
+ * Copyright (c) 2011 Kostas Spyropoulos <inigo.aldana@gmail.com>                       *
+ * Copyright (c) 2014 Bruno Romero de Azevedo <brunodea@inf.ufsm.br>                    *
+ *                                                                                      *
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/FlagToDisplay.kt:4]: Needs your copyright

**Review by david-allison — CHANGES_REQUESTED** [review]
One nitpick, then looks good to me, thanks!

File: AnkiDroid/src/main/java/com/ichi2/anki/FlagToDisplay.kt (hunk 1 above)
Comment (taulpi) [reply AnkiDroid/src/main/java/com/ichi2/anki/FlagToDisplay.kt]: Of course! Thank you for the fast feedback!

File: AnkiDroid/src/test/java/com/ichi2/anki/FlagToDisplayTest.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import com.ichi2.anki.reviewer.CardMarker.Companion.FLAG_ORANGE
+import com.ichi2.anki.reviewer.CardMarker.Companion.FLAG_PINK
+import com.ichi2.anki.reviewer.CardMarker.Companion.FLAG_PURPLE
+import com.ichi2.anki.reviewer.CardMarker.Companion.FLAG_RED
+import com.ichi2.anki.reviewer.CardMarker.Companion.FLAG_TURQUOISE
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ValueSource
+
+class FlagToDisplayTest {
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/FlagToDisplayTest.kt:16]: Especially: Awesome job here: tests are /much/ faster as you've avoided using Robolectric

**Review by david-allison — APPROVED** [review]
Very high quality, thanks for the fast response!

**Review by mikehardy — APPROVED** [review]
Really nice!

**Pull request merged**